src/sparsezoo/objects/model_folder_objects.py

```diff
@@ -0,0 +1,452 @@
+        except Exception as error:  # noqa: F841
+            self._throw_error(
+                error_msg="Json file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_csv(self, strict_mode):
+        try:
+            with open(self.path) as file:
+                file.readlines()
+        except Exception as error:  # noqa: F841
+            self._throw_error(
+                error_msg="Csv file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_img(self, strict_mode):
+        if not Image.open(self.path):
+            self._throw_error(
+                error_msg="Image file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _throw_error(self, error_msg, strict_mode):
+        if strict_mode:
+            raise ValueError(error_msg)
+        else:
+            logging.warning(error_msg)
+
+
+class Directory(File):
+    """
+    Object that represents a directory.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        files: Optional[List[File]] = None,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+        """
+        Constructor for a Directory object.
+
+        :param files: list of files contained within the Directory
+        :param name: name of the Directory
+        :param path: path of the Directory
+        :param url: url of the Directory
+        """
+        self.files = files
+
+        if path is None:
+            path = self._infer_path_from_files()
+        super().__init__(name=name, path=path, url=url)
+
+    def gzip(self) -> str:
+        """
+        Create a tar archive file from the Directory.
+        The tar archive file would contain all the files
+        in the Directory object.
+        The tar archive file would be saved in the parent
+        directory of the Directory object.
+        :return: path to the tar archive file created
+        """
+        if self.path is None:
+            raise ValueError(
+                "Attempting to zip the Directory object files, "
+                "but `self.path` is None. "
+                "Class object requires pointer to parent "
+                "folder directory to know where to save the tar archive file."
+            )
+        parent_path = pathlib.PurePath(self.path).parent
+        tar_file_path = os.path.join(parent_path, self.name + ".tar.gz")
+        with tarfile.open(tar_file_path, "w") as tar:
+            for file in self.files:
+                tar.add(file.path)
+        return tar_file_path
+
+    def unzip(self, path_to_extract: Optional[str] = None) -> bool:
+        """
+        If Directory is a tar archive, extract it to:
+         - parent directory of the Directory object (by default)
+         - specified path (if `tar_file_path` provided).
+
+        Note: `path_to_extract` is relative to the working dir!
```

Comment:
Let's talk about `tar.extractall()`. Either I did not understand properly its purpose or it is a super misleading piece of code.